discovery-provider/src/tasks/generate_trending.py

```diff
@@ -76,16 +77,32 @@ def generate_trending(db, time, genre, limit, offset):
             )
             .all()
         )
-        not_deleted_track_ids = set([record[0] for record in not_deleted_track_ids]) # pylint: disable=R1718
 
+        # Generate track -> created_at date
+        track_created_at_dict = {
+            record[0]: record[1] for record in not_deleted_track_ids
+        }
+
+        not_deleted_track_ids = set([record[0] for record in not_deleted_track_ids]) # pylint: disable=R1718
         # Query repost counts
         repost_counts = get_repost_counts(session, False, True, not_deleted_track_ids, None)
+         # Generate track_id --> repost_count mapping
         track_repost_counts = {
             repost_item_id: repost_count
             for (repost_item_id, repost_count, repost_type) in repost_counts
             if repost_type == RepostType.track
         }
 
+        # Query repost count with respect to rolling time frame in URL (e.g. /trending/week -> window = rolling week)
+        track_repost_counts_with_time = \
```

Comment:
similar request here - this object doesn't actually contain a new time variable, it is just track_repost_counts_for_time.

Reply:
looks good